Checkout/Source/Tokenisation/CheckoutAPIService.swift

```diff
@@ -187,6 +182,47 @@ final public class CheckoutAPIService: CheckoutAPIProtocol {
     }
   }
 
+  private func callRiskSDK(tokenDetails: TokenDetails,
+                           completion: @escaping () -> Void) {
+
+    /* Risk SDK calls can be finalised in 3 different ways
+     1. When Risk SDK's configure(...) function completed successfully and publishData(...) completed successfully or not
+     2. When Risk SDK's configure(...) function completed with failure
+     3. When Risk SDK's configure(...) or publishData(...) functions hang and don't call their completion blocks.
+        In this case, we wait for `self.timeoutInterval` amount of time and call the completion block anyway.
+
+     All these operations are done synchronously to avoid the completion closure getting called multiple times.
+    */
+
+    let finaliseRiskSDKCalls = {
+      self.taskCompletionQueue.sync {
+        if !self.isTaskCompleted {
+          self.isTaskCompleted = true
+          completion()
+        }
+      }
+    }
+
+    DispatchQueue.global(qos: .userInitiated).asyncAfter(deadline: .now() + timeoutInterval) {
+      finaliseRiskSDKCalls()
```

Comment:
Thread Safety: Ensure access to `isTaskCompleted` is protected by the `taskCompletionQueue` to prevent race conditions
```suggestion
     self.taskCompletionQueue.sync {
            finaliseRiskSDKCalls()
        }
```

Reply:
`finaliseRiskSDKCalls` closure has that check already